File: frontend/src/api/stats.ts
import { apiFetch } from "./client";

export type MostPlayedGame = {
  board_game_id: number;
  name: string;
  session_count: number;
};

export type StatsSummary = {
  total_sessions: number;
  total_play_time_minutes: number;
  average_duration_minutes: number | null;
  win_count: number;
  win_rate: number | null;
  sessions_last_30_days: number;
  most_played_games: MostPlayedGame[];
};

export async function fetchStatsSummary(): Promise<StatsSummary> {
  return (await apiFetch("/stats/summary")) as StatsSummary;
}

File: backend/app/schemas/stats.py
"""Schemas for dashboard statistics endpoints."""

from pydantic import BaseModel


class MostPlayedGame(BaseModel):
    """One game entry in the top played list."""

    board_game_id: int
    name: str
    session_count: int


class StatsSummary(BaseModel):
    """Aggregated per-user dashboard metrics."""

    total_sessions: int
    total_play_time_minutes: int
    average_duration_minutes: float | None
    win_count: int
    win_rate: float | None
    sessions_last_30_days: int
    most_played_games: list[MostPlayedGame]
